tool/vapid.c

```diff
@@ -0,0 +1,381 @@
+#include <stdlib.h>
+#include <string.h>
+
+#include <openssl/bn.h>
+#include <openssl/ec.h>
+#include <openssl/obj_mac.h>
+#include <openssl/sha.h>
+
+#include <ece.h>
+#include <ece/keys.h>
+
+#define VAPID_HEADER "{\"alg\":\"ES256\",\"typ\":\"JWT\"}"
+#define VAPID_HEADER_LENGTH 27
+
+static const char vapid_hex_table[] = "0123456789abcdef";
```

Comment:
Consider moving this into the library, now that `keygen` needs hex encoding, too.